### PR TITLE
feat: branch creation annotation support

### PR DESCRIPTION
**Pull request by antelman107**
resolves https://github.com/neondatabase/cloud/issues/15753

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,10 +54,11 @@
     "vitest": "^1.6.0"
   },
   "dependencies": {
-    "@neondatabase/api-client": "1.9.0",
+    "@neondatabase/api-client": "^1.10.0",
     "@segment/analytics-node": "^1.0.0-beta.26",
     "axios": "^1.4.0",
     "axios-debug-log": "^1.0.0",
+    "bun": "^1.1.21",
```
Comment (duskpoet) [inline package.json:61]: don't add bun to necessary dependencies, it's not required to run `neonctl`

File: src/commands/branches.ts [hunk 2]
```diff
@@ -100,6 +100,11 @@ export const builder = (argv: yargs.Argv) =>
             describe: 'Connect to a new branch via psql',
             default: false,
           },
+          annotation: {
+            type: 'string',
+            hidden: true,
```
Comment (duskpoet) [inline src/commands/branches.ts:105]: why hidden? we will not promote this option?

File: package.json (hunk 1 above)
Comment (antelman107) [reply package.json]: fixed

File: src/commands/branches.ts (hunk 2 above)
Comment (antelman107) [reply src/commands/branches.ts:105]: I would not promote it at least until we have all the parts are updated and tested on prod.

**Pull request merged**